src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala

```diff
@@ -1199,7 +1199,7 @@ abstract class SpecializeTypes extends InfoTransform with TypingTransformers {
    *  If it is a 'no-specialization' run, it is applied only to loaded symbols.
    */
   override def transformInfo(sym: Symbol, tpe: Type): Type = {
-    if (settings.nospecialization && currentRun.compiles(sym)) tpe
```

Comment:
The backend needs to specialize `TupleN` to populate its list of tuple constructors to make tests for box-elimination to work. This seems to be the only case covered by our test suite that requires specialization of a type after the specialization phase itself has run.